gd32e10x/cmsis/gd/gd32e10x/include/gd32e10x.h

```diff
@@ -94,7 +94,8 @@ OF SUCH DAMAGE.
 
 /* configuration of the Cortex-M4 processor and core peripherals */
 #define __CM4_REV                 0x0001   /*!< Core revision r0p1                                       */
-#define __MPU_PRESENT             0        /*!< GD32E10x do not provide MPU                              */
+// todo: change __MPU_PRESENT to 0
+#define __MPU_PRESENT             1        /*!< GD32E10x do not provide MPU                              */
```

Comment:
why is this required?

Reply:
I use this command to check header include
```
/opt/zephyr-sdk-0.13.1/arm-zephyr-eabi/bin/arm-zephyr-eabi-gcc -H -E -DBUILD_VERSION=v2.7.99-2019-gbe7336a1ef94 -DGD32E10X -DKERNEL -D_FORTIFY_SOURCE=2 -D__PROGRAM_START -D__ZEPHYR_SUPERVISOR__ -D__ZEPHYR__=1 -I/home/feilong/Program-ext/zephyr/zephyr/include -I/home/feilong/Program-ext/zephyr/zephyr/build/zephyr/include/generated -I/home/feilong/Program-ext/zephyr/zephyr/soc/arm/gigadevice/gd32e10x -I/home/feilong/Program-ext/zephyr/zephyr/soc/arm/gigadevice/common/. -I/home/feilong/Program-ext/zephyr/zephyr/soc/arm/gigadevice/gd32e10x/. -I/home/feilong/Program-ext/zephyr/modules/hal/cmsis/CMSIS/Core/Include -I/home/feilong/Program-ext/zephyr/modules/hal/gigadevice/gd32e10x/cmsis/gd/gd32e10x/include -I/home/feilong/Program-ext/zephyr/modules/hal/gigadevice/gd32e10x/standard_peripheral/include -I/home/feilong/Program-ext/zephyr/modules/hal/gigadevice/include -isystem /home/feilong/Program-ext/zephyr/zephyr/lib/libc/minimal/include -isystem /opt/zephyr-sdk-0.13.1/arm-zephyr-eabi/bin/../lib/gcc/arm-zephyr-eabi/10.3.0/include -isystem /opt/zephyr-sdk-0.13.1/arm-zephyr-eabi/bin/../lib/gcc/arm-zephyr-eabi/10.3.0/include-fixed -Os -imacros /home/feilong/Program-ext/zephyr/zephyr/build/zephyr/include/generated/autoconf.h -ffreestanding -fno-common -g -gdwarf-4 -fdiagnostics-color=always -mcpu=cortex-m4 -mthumb -mabi=aapcs -mfp16-format=ieee -imacros /home/feilong/Program-ext/zephyr/zephyr/include/toolchain/zephyr_stdint.h -Wall -Wformat -Wformat-security -Wno-format-zero-length -Wno-main -Wno-pointer-sign -Wpointer-arith -Wexpansion-to-defined -Wno-address-of-packed-member -Wno-unused-but-set-variable -Werror=implicit-int -fno-asynchronous-unwind-tables -fno-pie -fno-pic -fno-strict-overflow -fno-reorder-functions -fno-defer-pop -fmacro-prefix-map=/home/feilong/Program-ext/zephyr/zephyr/samples/basic/minimal=CMAKE_SOURCE_DIR -fmacro-prefix-map=/home/feilong/Program-ext/zephyr/zephyr=ZEPHYR_BASE -fmacro-prefix-map=/home/feilong/Program-ext/zephyr=WEST_TOPDIR -ffunction-sections -fdata-sections -std=c99 -nostdinc -o build/e103.i -c /home/feilong/Program-ext/zephyr/zephyr/drivers/pinctrl/pinctrl_gd32_afio.c 2>build/header.txt
```

then I find the include chain of hal_gigadevice is:
```
. /home/feilong/Program-ext/zephyr/zephyr/include/drivers/pinctrl.h
.. /home/feilong/Program-ext/zephyr/zephyr/include/device.h
... /home/feilong/Program-ext/zephyr/zephyr/include/init.h
.... /home/feilong/Program-ext/zephyr/zephyr/include/kernel.h
..... /home/feilong/Program-ext/zephyr/zephyr/include/kernel_includes.h
...... /home/feilong/Program-ext/zephyr/zephyr/include/arch/cpu.h
....... /home/feilong/Program-ext/zephyr/zephyr/include/arch/arm/aarch32/arch.h
........ /home/feilong/Program-ext/zephyr/zephyr/include/arch/arm/aarch32/mpu/arm_mpu.h
......... /home/feilong/Program-ext/zephyr/zephyr/include/arch/arm/aarch32/mpu/arm_mpu_v7m.h
.......... /home/feilong/Program-ext/zephyr/zephyr/include/arch/arm/aarch32/cortex_m/cmsis.h
........... /home/feilong/Program-ext/zephyr/zephyr/soc/arm/gigadevice/gd32e10x/soc.h
............ /home/feilong/Program-ext/zephyr/modules/hal/gigadevice/gd32e10x/cmsis/gd/gd32e10x/include/gd32e10x.h
............. /home/feilong/Program-ext/zephyr/modules/hal/gigadevice/gd32e10x/standard_peripheral/include/gd32e10x_libopt.h
.............. /home/feilong/Program-ext/zephyr/modules/hal/gigadevice/gd32e10x/standard_peripheral/include/gd32e10x_gpio.h
.............. /home/feilong/Program-ext/zephyr/modules/hal/gigadevice/gd32e10x/standard_peripheral/include/gd32e10x_usart.h

```

because of the code in https://github.com/zephyrproject-rtos/zephyr/blob/aceed99e2d224960d398fd812032ba71eba10162/include/arch/arm/aarch32/arch.h#L187-L197
if I don't enable MPU, It will not include any header file in `hal_gigadevice`.

Reply:
fixed by https://github.com/zephyrproject-rtos/hal_gigadevice/pull/12#issuecomment-991413500